Initiate VuePress documentation

Shinigami92 opened the pull request:
closes #188 

**Demo:** https://prettier.github.io/plugin-pug

## TODOs

- [x] Link to GitHub Repo
- [x] Getting started on own site
- [x] Site for every rule with examples
  - standard prettier rules:
    - [x] pugPrintWidth
    - [x] pugSingleQuote
    - [x] pugTabWidth
    - [x] pugUseTabs
    - [x] pugBracketSpacing
    - [x] pugArrowParens
    - [x] pugSemi
  - pug-specific rules:
    - [x] pugAttributeSeparator
    - [x] pugClosingBracketPosition
    - [x] pugCommentPreserveSpaces
    - [x] pugSortAttributes
    - [x] pugSortAttributesBeginning
    - [x] pugSortAttributesEnd
    - [x] pugWrapAttributesThreshold
    - [x] pugWrapAttributesPattern
    - [x] pugClassNotation
    - [x] pugIdNotation
    - [x] pugEmptyAttributes
    - [x] pugEmptyAttributesForceQuotes
    - [x] pugSingleFileComponentIndentation
- [ ] Workarounds on own site
- [x] Provide Dark Theme
- [x] Compact the repo `README.md`
- [x] Add link in repo `README.md` to docu site

Shinigami92 commented:
@SkyaTura @kaangokdemir Are you willing to give me a first feedback?
There is currently 4 things:
- [Home](https://prettier.github.io/plugin-pug/)
- [Guide](https://prettier.github.io/plugin-pug/guide/)
- [Configuration](https://prettier.github.io/plugin-pug/configuration/)
- [pugAttributeSeparator](https://prettier.github.io/plugin-pug/configuration/pug-specific-options/pugAttributeSeparator.html#description)

kaangokdemir commented on docs/guide/README.md:
````diff
@@ -0,0 +1,86 @@
+Pug code with preserved custom formatting:
+
+<!-- prettier-ignore -->
+```pug
````
<img width="951" alt="Ekran Resmi 2021-03-29 20 40 40" src="https://user-images.githubusercontent.com/51219535/112883810-087d6e00-90cf-11eb-9f07-d4feef5e34e3.png">

For some reason, backtick's are displayed on render.

Shinigami92 replied:
This is expected :smiley:  
It's a markdown embedded in markdown!

kaangokdemir reviewed (commented):
Looking good to me.

Shinigami92 commented:
@patak-js and others I just added via request review

Everyone is welcome to explore the new [documentation](https://prettier.github.io/plugin-pug) and give me feedback :open_hands:

patak-js commented:
Hey @Shinigami92, this looks awesome! That home page is sweet :)

A few ideas to throw on the table, in case you find them useful

- In Guide, I would remove the "Configuration" title. Users can click already in the navbar, or when they reach the bottom with the next button
- In Configuration, I would prefer that the pug-specific options would leave the sidebar with the three levels so I can explore what each option does with one click (instead of two). I think it helps with discoverability
- For Windows users, in VueUse, we ended up styling the scrollbars, because the default ones are not that pretty. Here is the PR for that https://github.com/vueuse/vueuse/pull/275. In this case, it is just a detail, because the sidebar menu doesn't need one, so just a heads up in case the sidebar grows at one point.
- Looks like the favicon is not kicking in
- There is an option to add Edit this page, that may be useful to enable in case you would like to push users to send you docs PRs

Shinigami92 commented:
@patak-js Would you like to have a new look into the docs?
I now changed the sidebar structure and the specific options are now all on one page so the user can e.g. search with `(ctrl|cmd)+f`
I also modified `hr` tags to be a bit more thick so hopefully users can see sections better

patak-js commented:
Looks awesome @Shinigami92! ❤️

ST-DDT commented on docs/guide/pug-specific-options.md:
````diff
@@ -0,0 +1,453 @@
+Allow one unwrapped attribute, wrap two and more.
+
+```pug
+input(type="text")
````
Maybe also add an example with two attributes here. Because it is not clear whether the wrapping starts with option=attributes or option=attributes+1

Or maybe change the description a bit:

`Allow up to one attribute to remain in the same line. Wrap all attributes, if there are two or more attributes.`

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,453 @@
+### Options
+
+**Type:** `choice`  
+**Default:** `'none'`
```
Probably off topic, but maybe `auto` or `detect` would be a better value for "auto detect".

Shinigami92 replied:
I also thought about that, but it would be a code change instead of a docs change

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,453 @@
+
+### Description
+
+coming soon
```
TODO

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,453 @@
+
+#### `'none'`
+
+Every attribute with empty quotes will have them removed.
```
```suggestion
Every boolean attribute with empty quotes will have them removed.
```

Or does this apply to all attributes?

```suggestion
Every attribute (regardless of type) with empty quotes will have them removed.
```

Shinigami92 replied:
This doesn't make sense theoretically cause HTML attributes are not typed at all

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,453 @@
+**Type:** `choice`  
+**Default:** `'as-is'`
+
+Change behavior of boolean attributes.
```
Maybe add a hint that there is also the `pugEmptyAttributesForceQuotes` rule.

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,453 @@
+
+### Description
+
+Define a list of patterns for attributes that will be forced to have empty quotes even with "none" selected.
```
```suggestion
Define a list of patterns for attributes that will be forced to have empty quotes even with "pugEmptyAttributes=none" selected.
```

ST-DDT commented on docs/guide/pug-specific-options.md:
````diff
@@ -0,0 +1,453 @@
+```
+
+::: warning
+This option is not yet available  
````
Available or implemented?

ST-DDT commented on docs/guide/pug-specific-options.md:
````diff
@@ -0,0 +1,453 @@
+```
+
+::: warning
+This option is not yet available  
````
Available or implemented?

ST-DDT commented on docs/index.md:
```diff
@@ -0,0 +1,46 @@
+footer: MIT Licensed | Copyright © 2019-2021 Christopher Quadflieg
+---
+
+<!-- TODO christopher 2021-03-29: Move this to heroImage -->
```
TODO

Shinigami92 replied:
Only possible with VuePress v2 or maybe VitePress

Shinigami92 commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,453 @@
+
+### Options
+
+**Type:** `array`  
```
It's already mentioned in the sentence directly above this

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,472 @@
+
+#### `'new-line'`
+
+Closing bracket ends with a new line.
```
```suggestion
The closing bracket is placed on a new line.
```

ST-DDT commented on docs/guide/pug-specific-options.md:
```diff
@@ -0,0 +1,472 @@
+
+#### `'last-line'`
+
+Closing bracket remains with last attribute's line.
```
```suggestion
The closing bracket is placed directly after the last attribute without an additional line break.
```
